src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -853,6 +856,12 @@ public IEnumerable<string> GetCollections(Transaction transaction)
                     }
 
                     yield return revision;
+                    if (first)
+                    {
+                        first = false;
+                        skip++;
+                    }
```

Comment:
This seems like a bug not related to this issue, right?
Do you have a failing test for this case? Maybe we should fix in a separate PR?

Reply:
No. it is not a bug.
In the previous code, you deleted and then got the first revision, then you deleted it and again got the first revision..
In the new code, you are not delete the last you got so you need to skip it in every interaction of the DeleteRevoisionsInternal.

Reply:
Okay, I understand now.
Let's talk about this.. maybe would be better to pass this information over the `DeleteOldRevisionsResult` object.

Reply:
Done
The Skip now comes from the 'result' object..